### PR TITLE
Fix header issue

### PR DESCRIPTION
**Pull request by kevdez**
- add file structure
- use `classnames` tool
- `emotion` for styled components, css-in-js
- improved validation for pokemon input field by lower-casing everything. Ideally a fuzzy search list would be cool.

## Bugfix:

When you're on the Pokemon page, search for a pokemon and then refresh the page: the `Home` button is highlighted in the nav. The expected button to be highlighted should be `Pokemon`.

You were using React state to keep track of the active page, but it's better to use react-router, since it already gives you that data. 

Screen Shot 2019-05-05 at 9 55 23 PM

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -13,73 +14,35 @@ function Berries() { return <h2> Berries page </h2>; }
...
-      var idStr = curr.idStr;
-      var classStr = ( (idStr === this.state.activeLink) ? "active-link": ""); 
-
-      return (
-        <li className={classStr}
-            id={curr.idStr}
-            key={curr.idStr}
-            onClick={(e) => this.handleLinkClick(e)}>
-
-          <Link to={curr.url}> {curr.text} </Link>
```
Comment (kevdez) [inline client/src/App.js:51]: Just as an FYI, `<Link>`s are already `<a>` elements. So when you were clicking on this button, you were clicking on Link (`<a>`'s href) not on `<li>`'s `onClick`. So old lines 46-49 were unnecessary.

File: client/src/App.js [hunk 2]
```diff
@@ -13,73 +14,35 @@ function Berries() { return <h2> Berries page </h2>; }
...
+          <Route path="/Items" component={Items} />
+          <Route path="/Berries" component={Berries} />
+          <Route component={NoMatch} />
+        </Switch> 
+
+        <footer className="App-footer">
+          Created using <a href="https://reactjs.org" target="_blank" rel="noopener noreferrer">React</a>.
+          Uses the <a href="https://pokeapi.co" target="_blank" rel="noopener noreferrer">Pok&eacute;API.</a>
+        </footer>
+      </div>
```
Comment (kevdez) [inline client/src/App.js:45]: this whole change is just tabbing everything over 2 spaces.
Your nesting had one case where it was tabbed 4 spaces, and the rest of the file's style was 2-space

File: client/src/components/nav-links.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  renderLinks() {
+    return LINKS.map(({ url, text }, i) => {
+      const {
+        location: { pathname }
+      } = this.props;
+      const shouldHighlightLink =
+        pathname === url || (i > 0 && pathname.includes(url));
+      return (
+        <Link className={cx({ "active-link": shouldHighlightLink })} to={url}>
```
Comment (kevdez) [inline client/src/components/nav-links.jsx:32]: `classnames` in action as `cx`. https://jedwatson.github.io/classnames/

File: client/src/components/nav-links.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          {text}
+        </Link>
+      );
+    });
+  }
+
+  render() {
+    return (
+      <nav>
+        <ULFlex>{this.renderLinks()}</ULFlex>
```
Comment (kevdez) [inline client/src/components/nav-links.jsx:42]: `@emotion/styled` components in action. The whole `emotion` toolset is pretty good for CSS, so I added it to this project.

File: client/src/pokemon/PokemonInfo.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+  componentDidUpdate(prevProps) {
+    const previousName = prevProps.match.params.pokemon;
+    const {
+      match: {
+        params: {
+          pokemon
+        }
+      }
+    } = this.props;
```
Comment (kevdez) [inline client/src/pokemon/PokemonInfo.js:47]: next level destructuring lmao
Normally this is used for destructuring both sibling and nested props together. In this one case, a one-liner would've been fine.

```js
const {pokemon} = this.props.match.params;
```

File: server.js [hunk 6]
```diff
@@ -19,19 +19,18 @@ app.get('/getPokemon/:name', (req, res) => {
...
-      console.log(`GET /pokemon/${name}`); 
-      res.send(response); 
+      console.log(`GET /pokemon/${name}`);
+      res.send(response);
     })
     .catch(function(){
-      res.send(null); 
-      console.log(`ERROR AT: GET /getPokemon/${name}`); 
-      console.log(`\t Name: ${name}`); 
-    })
```
Comment (kevdez) [inline server.js:29]: Prettier in action, catching weird spaces from my VSCode plugin, without a config file. This entire file was fixed automatically, as a demo.

File: client/src/App.js [hunk 7]
```diff
@@ -13,73 +14,35 @@ function Berries() { return <h2> Berries page </h2>; }
...
-          <footer className="App-footer">
-            Created using <a href="https://reactjs.org" target="_blank" rel="noopener noreferrer">React</a>.
-            Uses the <a href="https://pokeapi.co" target="_blank" rel="noopener noreferrer">Pok&eacute;API.</a>
-          </footer>
-        </div>
-      </Router>
+      <div className="App">
+        <header className="App-header">
+          <h1> Pok&eacute;mon Info App </h1> 
+          <Route component={NavLinks} />
```
Comment (kevdez) [inline client/src/App.js:29]: All your header Link logic was moved into the NavLinks component + also fixing the bug.

**Comment by kevdez** [comment]
Screen Shot 2019-05-05 at 9 55 23 PM

**Comment by rodrigih** [comment]
Thanks for the help!

**Pull request merged**